cmd/pod-scaler/frontend/src/app/CircLLHist/LogarithmicComparativePlot.tsx

```diff
@@ -0,0 +1,403 @@
+import * as React from 'react';
+import {Flex, FlexItem, Slider, Text, TextContent, TextVariants} from '@patternfly/react-core';
+import {Bin, Histogram, Value, ValueBounds, Width} from "@app/CircLLHist/CircLLHist";
+import * as d3 from 'd3-scale-chromatic';
+
+export interface LogarithmicComparativePlotProps {
+  /** the cutoff to display on the plot */
+  cutoff: number;
+  /** the lower bound for display on the plot */
+  lower_bound: number;
+  /** the merged data to show as a summary on the left of the plot */
+  merged: Histogram;
+  /** the data to plot */
+  histograms: Histogram[];
+  /** options for the canvas layout */
+  canvasProps: CanvasProps & OptionalCanvasProps;
+}
+
+export interface CanvasProps {
+  /** title of the plot */
+  title: string;
+  /** title of the Y axis */
+  yAxisTitle: string;
+  /** units on the Y axis */
+  yAxisUnit: string;
+  /** minimum value for the Y axis */
+  yAxisMin: number;
+  /** formatter for Y axis ticks */
+  yAxisFormatter: (value: number) => string;
+}
+
+export interface OptionalCanvasProps {
+  /** horizontal offset for the plot, in pixels */
+  sideOffset?: number;
+  /** vertical offset for the plot, in pixels */
+  verticalOffset?: number;
+  /** extra offset for the top of the plot, in pixels */
+  topOffset?: number;
+  /** width of a cell, in pixels */
+  cellWidth?: number;
+  /** height of the data in the plot, in pixels */
+  dataHeight?: number;
+}
+
+const defaultSideOffset = 100;
+const defaultVerticalOffset = 20;
+const defaultTopOffset = 50;
+const defaultCellWidth = 20;
+const defaultDataHeight = 1000;
+
+
+type CanvasMetadata = CanvasProps & ProvidedMetadata & ComputedMetadata;
+
+interface ProvidedMetadata {
+  /** horizontal offset for the plot, in pixels */
+  sideOffset: number;
+  /** vertical offset for the plot, in pixels */
+  verticalOffset: number;
+  /** extra offset for the top of the plot, in pixels */
+  topOffset: number;
+  /** width of a cell, in pixels */
+  cellWidth: number;
+  /** height of the data in the plot, in pixels */
+  dataHeight: number;
+}
+
+interface ComputedMetadata {
+  /** the number of columns in the plot */
+  numCols: number;
+  /** the width of the plot, in pixels */
+  width: number;
+  /** the height of the plot, in pixels */
+  height: number;
+
+  /** the most negative value in the plot */
+  minimum: number;
+  /** the most positive value in the plot */
+  maximum: number;
+
+  /** state stored for ID generation */
+  nextId: number;
+}
+
+const canvasMetaFromProps = (props: LogarithmicComparativePlotProps, windowWidth?: number): CanvasMetadata => {
+  const canvasProps: OptionalCanvasProps = props.canvasProps as OptionalCanvasProps ? props.canvasProps as OptionalCanvasProps : {};
```

Comment:
Typically if you need to use `as`, you have an incorrect type definition somewhere. It's good to avoid in general.